Git Streak Tracker/SettingsView.swift

```diff
@@ -10,29 +10,77 @@ import SwiftUI
 import WidgetKit
 import SwiftUIFontIcon
 
+
 struct SettingsView: View {
+    var onContributionDataFetched: ((ContributionData) -> Void)? = nil
+    
     @EnvironmentObject private var userStore: UserStore
     @State private var inputValue: String = ""
     @State private var isEditing = false
-    
-    let debouncer = Debouncer(delay: 0.5)
-    
+        
+    // Actions
     func loadInputField() {
         inputValue = userStore.username
     }
-    
-    func storeUsername() {
-        userStore.username = inputValue // everytime username changes, the contributions are requested
+        
+    func handleSaveUsernamePress() {
+        // everytime username changes, the contributions are requested
+        inputValue = inputValue.trimmingCharacters(in: .whitespacesAndNewlines)
+        userStore.setUsername(
+            username: inputValue,
+            onComplete: onContributionDataFetched
+        )
         WidgetCenter.shared.reloadAllTimelines()
     }
     
+    // Computed
     func getShadeColor() -> Color{
         if userStore.contributionData.error {
             return ColorPallete.midRed
         }
         return isEditing ? ColorPallete.highlightGreen : .clear
     }
     
+    func isUsernameInputValid() -> Bool {
+        return inputValue.trimmingCharacters(in: .whitespacesAndNewlines) != ""
+    }
+    
+    func getSaveUsernameBg() -> LinearGradient {
+        if (!isUsernameInputValid()) {
+            return LinearGradient(
+                gradient: Gradient(colors: [ColorPallete.midGreen, ColorPallete.midGreen]),
+                startPoint: .leading,
+                endPoint: .trailing
+            )
+        }
+        
+        return LinearGradient(
+            gradient: Gradient(colors: [ColorPallete.greenGradientStart, ColorPallete.greenGradientEnd]),
+            startPoint: .leading,
+            endPoint: .trailing
+        )
+    }
```

Comment:
Maybe i'm nit picking, but this is a bit cleaner imo

```
func getSaveUsernameBg() -> LinearGradient {
    let gradientColors = isUsernameInputValid() ?
        [ColorPallete.greenGradientStart, ColorPallete.greenGradientEnd] :
        [ColorPallete.midGreen, ColorPallete.midGreen]
    
    return LinearGradient(
        gradient: Gradient(colors: gradientColors),
        startPoint: .leading,
        endPoint: .trailing
    )
}
```

Reply:
But from your code I kinda get the feeling that you don't like ternaries, eh?

Reply:
Yeah I like big if blocks these days haha. I'll match this repos style though and update.

Reply:
I think whether you use a ternary or an if statement doesn't matter, but should just operate on that array of colors to keep it DRY.